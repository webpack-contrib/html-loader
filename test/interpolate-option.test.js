import {
  compile,
  execute,
  getCompiler,
  getErrors,
  getModuleSource,
  getWarnings,
  readAsset,
} from './helpers';

describe("'interpolate' option", () => {
  it('should disabled by default', async () => {
    const compiler = getCompiler('template.js');
    const stats = await compile(compiler);

    expect(getModuleSource('./template.html', stats)).toMatchSnapshot('module');
    expect(
      execute(readAsset('main.bundle.js', compiler, stats))
    ).toMatchSnapshot('result');
    expect(getWarnings(stats)).toMatchSnapshot('warnings');
    expect(getErrors(stats)).toMatchSnapshot('errors');
  });

  it('should work with boolean notation', async () => {
    const compiler = getCompiler('template.js', { interpolate: true });
    const stats = await compile(compiler);

    expect(getModuleSource('./template.html', stats)).toMatchSnapshot('module');
    expect(
      execute(readAsset('main.bundle.js', compiler, stats))
    ).toMatchSnapshot('result');
    expect(getWarnings(stats)).toMatchSnapshot('warnings');
    expect(getErrors(stats)).toMatchSnapshot('errors');
  });

  it('should work with the "require"', async () => {
    const compiler = getCompiler('template.js', { interpolate: 'require' });
    const stats = await compile(compiler);

    expect(getModuleSource('./template.html', stats)).toMatchSnapshot('module');
    expect(
      execute(readAsset('main.bundle.js', compiler, stats))
    ).toMatchSnapshot('result');
    expect(getWarnings(stats)).toMatchSnapshot('warnings');
    expect(getErrors(stats)).toMatchSnapshot('errors');
  });

  it('should emit an error on broken interpolation syntax', async () => {
    const compiler = getCompiler('broken-interpolation-syntax.js', {
      interpolate: true,
    });
    const stats = await compile(compiler);

    expect(
      getModuleSource('./broken-interpolation-syntax.html', stats)
    ).toMatchSnapshot('module');
    expect(
      execute(readAsset('main.bundle.js', compiler, stats))
    ).toMatchSnapshot('result');
    expect(getWarnings(stats)).toMatchSnapshot('warnings');
    expect(getErrors(stats)).toMatchSnapshot('errors');
  });
});
