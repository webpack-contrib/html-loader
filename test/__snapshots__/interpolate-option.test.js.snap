// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`'interpolate' option should disabled by default: errors 1`] = `Array []`;

exports[`'interpolate' option should disabled by default: module 1`] = `
"// Exports
module.exports = \\"<img src=\\\\\\"\${\\\\\\"Hello \\\\\\" + (1+1)}\\\\\\">\\\\n<img src=\\\\\\"\${\`Hello \` + (1+1)}\\\\\\">\\\\n<p>Something about the \\\\\\\\\` character</p>\\\\n<script>{\\\\\\"json\\\\\\": \\\\\\"with \\\\\\\\\\\\\\"quotes\\\\\\\\\\\\\\" in value\\\\\\"}</script>\\\\n<h3>#{number} {customer}</h3>\\\\n<p>   {title}   </p>\\\\n<img src=\\\\\\"\${require(\\\\\\"./image.png\\\\\\")}\\\\\\" /></a>\\\\n\\";"
`;

exports[`'interpolate' option should disabled by default: result 1`] = `
"<img src=\\"\${\\"Hello \\" + (1+1)}\\">
<img src=\\"\${\`Hello \` + (1+1)}\\">
<p>Something about the \\\\\` character</p>
<script>{\\"json\\": \\"with \\\\\\"quotes\\\\\\" in value\\"}</script>
<h3>#{number} {customer}</h3>
<p>   {title}   </p>
<img src=\\"\${require(\\"./image.png\\")}\\" /></a>
"
`;

exports[`'interpolate' option should disabled by default: warnings 1`] = `Array []`;

exports[`'interpolate' option should emit an error on broken interpolation syntax: errors 1`] = `
Array [
  "ModuleError: Module Error (from /path/to/file.js):
Line 1: Unexpected token |",
]
`;

exports[`'interpolate' option should emit an error on broken interpolation syntax: module 1`] = `
"// Exports
module.exports = \\"<div>\${broken|||test}</div>\\\\n\\";"
`;

exports[`'interpolate' option should emit an error on broken interpolation syntax: result 1`] = `
"<div>\${broken|||test}</div>
"
`;

exports[`'interpolate' option should emit an error on broken interpolation syntax: warnings 1`] = `Array []`;

exports[`'interpolate' option should work with boolean notation: errors 1`] = `Array []`;

exports[`'interpolate' option should work with boolean notation: module 1`] = `
"// Exports
module.exports = \\"<img src=\\\\\\"\\" + (\\"Hello \\" + (1 + 1)) + \\"\\\\\\">\\\\n<img src=\\\\\\"\\" + (\\"Hello \\" + (1 + 1)) + \\"\\\\\\">\\\\n<p>Something about the \` character</p>\\\\n<script>{\\\\\\"json\\\\\\": \\\\\\"with \\\\\\\\\\\\\\"quotes\\\\\\\\\\\\\\" in value\\\\\\"}</script>\\\\n<h3>#{number} {customer}</h3>\\\\n<p>   {title}   </p>\\\\n<img src=\\\\\\"\\" + require(\\"./image.png\\") + \\"\\\\\\" /></a>\\\\n\\";"
`;

exports[`'interpolate' option should work with boolean notation: result 1`] = `
"<img src=\\"Hello 2\\">
<img src=\\"Hello 2\\">
<p>Something about the \` character</p>
<script>{\\"json\\": \\"with \\\\\\"quotes\\\\\\" in value\\"}</script>
<h3>#{number} {customer}</h3>
<p>   {title}   </p>
<img src=\\"[object Module]\\" /></a>
"
`;

exports[`'interpolate' option should work with boolean notation: warnings 1`] = `Array []`;

exports[`'interpolate' option should work with the "require": errors 1`] = `Array []`;

exports[`'interpolate' option should work with the "require": module 1`] = `
"// Imports
var ___HTML_LOADER_GET_URL_IMPORT___ = require(\\"../../src/runtime/getUrl.js\\");
var ___HTML_LOADER_IDENT_0___ = require(\\"./image.png\\");
// Exports
module.exports = \\"<img src=\\\\\\"\${\\\\\\"Hello \\\\\\" + (1+1)}\\\\\\">\\\\n<img src=\\\\\\"\${\`Hello \` + (1+1)}\\\\\\">\\\\n<p>Something about the \\\\\\\\\` character</p>\\\\n<script>{\\\\\\"json\\\\\\": \\\\\\"with \\\\\\\\\\\\\\"quotes\\\\\\\\\\\\\\" in value\\\\\\"}</script>\\\\n<h3>#{number} {customer}</h3>\\\\n<p>   {title}   </p>\\\\n<img src=\\\\\\"\\" + ___HTML_LOADER_GET_URL_IMPORT___(___HTML_LOADER_IDENT_0___) + \\"\\\\\\" /></a>\\\\n\\";"
`;

exports[`'interpolate' option should work with the "require": result 1`] = `
"<img src=\\"\${\\"Hello \\" + (1+1)}\\">
<img src=\\"\${\`Hello \` + (1+1)}\\">
<p>Something about the \\\\\` character</p>
<script>{\\"json\\": \\"with \\\\\\"quotes\\\\\\" in value\\"}</script>
<h3>#{number} {customer}</h3>
<p>   {title}   </p>
<img src=\\"/webpack/public/path/image.png\\" /></a>
"
`;

exports[`'interpolate' option should work with the "require": warnings 1`] = `Array []`;
